app/bundles/CoreBundle/Helper/BuilderTokenHelperFactory.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Mautic\CoreBundle\Helper;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\ORM\EntityManager;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+
+class BuilderTokenHelperFactory
+{
+    private $security;
+    private $entityManager;
+    private $connection;
+    private $userHelper;
+
+    /**
+     * @param CorePermissions $security
+     * @param EntityManager   $entityManager
+     * @param Connection      $connection
+     * @param UserHelper      $userHelper
+     */
+    public function __construct(
+        CorePermissions $security,
+        EntityManager $entityManager,
+        Connection $connection,
+        UserHelper $userHelper
+    ) {
+        $this->security      = $security;
+        $this->entityManager = $entityManager;
+        $this->connection    = $connection;
+        $this->userHelper    = $userHelper;
+    }
+
+    /**
+     * @param $modelName
+     * @param string $viewPermissionBase
+     * @param string $bundleName
+     * @param string $langVar
+     *
+     * @return BuilderTokenHelper
+     */
+    public function getBuilderTokenHelper($modelName, $viewPermissionBase = null, $bundleName = null, $langVar = null)
```

Comment:
Maybe we could talk about this, but I think that it would make sense to use param and return types for the new classes and methods from this point forward. I don't see any reason not to.
```suggestion
    public function getBuilderTokenHelper(string $modelName, ?string $viewPermissionBase = null, ?string $bundleName = null, ?string $langVar = null): BuilderTokenHelper
```

Reply:
I agree with this, I think we should take full of advantage of the strong type system.